File: lib/models/slot.dart
class Slot {
  final String id;
  final String venueId;
  final DateTime startTime;
  final DateTime endTime;
  final bool isBooked;
  final String? bookedByUserId;

  Slot({
    required this.id,
    required this.venueId,
    required this.startTime,
    required this.endTime,
    required this.isBooked,
    this.bookedByUserId,
  });

  factory Slot.fromMap(Map<String, dynamic> map) {
    return Slot(
      id: map['id'] as String,
      venueId: map['venueId'] as String,
      startTime: DateTime.parse(map['startTime'] as String),
      endTime: DateTime.parse(map['endTime'] as String),
      isBooked: map['isBooked'] as bool,
      bookedByUserId: map['bookedByUserId'] as String?,
    );
  }

  Map<String, dynamic> toMap() => {
    'id': id,
    'venueId': venueId,
    'startTime': startTime.toIso8601String(),
    'endTime': endTime.toIso8601String(),
    'isBooked': isBooked,
    'bookedByUserId': bookedByUserId,
  };
}

File: server/src/main/java/com/quickslot/model/Slot.java
package com.quickslot.model;

import jakarta.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "slots", uniqueConstraints = {@UniqueConstraint(columnNames = {"venue_id","start_time"})})
public class Slot {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "venue_id")
    private Long venueId;

    @Column(name = "start_time")
    private LocalDateTime startTime;

    @Column(name = "end_time")
    private LocalDateTime endTime;

    @Column(name = "is_booked")
    private boolean isBooked = false;

    @Column(name = "booked_by_user_id")
    private String bookedByUserId;

    public Slot() {}

    public Slot(Long venueId, LocalDateTime startTime, LocalDateTime endTime) {
        this.venueId = venueId;
        this.startTime = startTime;
        this.endTime = endTime;
    }

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public Long getVenueId() { return venueId; }
    public void setVenueId(Long venueId) { this.venueId = venueId; }
    public LocalDateTime getStartTime() { return startTime; }
    public void setStartTime(LocalDateTime startTime) { this.startTime = startTime; }
    public LocalDateTime getEndTime() { return endTime; }
    public void setEndTime(LocalDateTime endTime) { this.endTime = endTime; }
    public boolean isBooked() { return isBooked; }
    public void setBooked(boolean booked) { isBooked = booked; }
    public String getBookedByUserId() { return bookedByUserId; }
    public void setBookedByUserId(String bookedByUserId) { this.bookedByUserId = bookedByUserId; }
}
